2-types/part_1-2/src/part_2.rs

```diff
@@ -1,49 +1,83 @@
+use std::fs::File;
+use std::io::BufReader;
+use std::path::Path;
 
-// Part 2: Safer time handling with Option<Result> and pattern matching.
-// We model a device with optional local_time parsed from string.
-// We expose a method to render local_time, or return a descriptive error.
+use serde::{Deserialize, Serialize};
 
-#[derive(Debug, Clone)]
-pub struct Device {
-    pub id: String,
-    pub local_time: Option<String>,
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Request {
+    pub r#type: String,
```

Comment:
1. Always avoid `r#` syntax. It is confusing for others
2. You can type-check the request type


```suggestion
#[derive(Serialize, Deserialize, Debug, PartialEq)]
#[serde(rename_all = "lowercase")]
enum RequestType {
    Success,
    Fail,
}
    #[serde(rename = "type")]
    pub request_type: RequestType,
```